internal/services/iothub/iothub_dps_resource.go

```diff
@@ -139,6 +177,12 @@ func resourceIotHubDPS() *pluginsdk.Resource {
 				}, false),
 			},
 
+			"public_network_access_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  true,
```

Comment:
Adding new default values to existing resources can be tricky if the api doesn't return that attribute on already existing resources. I just checked by spinning up the resource on the current provider and it didn't show up in the api. I then tried the planned changes here against the old state and we got a plan that we shouldn't have. 

```
      ~ public_network_access_enabled = false -> true
```

I do see that this attribute is `true` in the portal when we provision the resource there so we should just be able to add a state migration to force this into people's state with the next provider version